src/select.ts

```diff
@@ -11,17 +10,42 @@ export class NgrxSelect {
   }
 }
 
-export function Select(path?: string): any {
-  return function(target: any, name: string, descriptor: TypedPropertyDescriptor<any>): void {
+export function Select<TState = any, TValue = any>(
+  selector: Selector<TState, TValue>
+): (target: any, name: string) => void;
+export function Select<TState = any, TValue = any>(
+  selectorOrFeature?: string,
+  ...paths: string[]
+): (target: any, name: string) => void;
+export function Select<TState = any, TValue = any>(
+  selectorOrFeature: string | Selector<TState, TValue>,
+  ...paths: string[]
+) {
+  return function(target: any, name: string): void {
+    let fn: Selector<TState, TValue>;
+    // Nothing here? Use propery name as selector
+    if (typeof selectorOrFeature === 'undefined') {
+      selectorOrFeature = name;
+    }
+    // Handle string vs Selector<TState, TValue>
+    if (typeof selectorOrFeature === 'string') {
+      if (paths.length) {
+        selectorOrFeature = [selectorOrFeature, ...paths].join('.');
+      }
+      fn = fastPropGetter(selectorOrFeature);
+    } else {
+      fn = selectorOrFeature;
+    }
```

Comment:
why do we have to `.join('.')`, then `.split('.')`. 
how about 
```ts
function fastPropGetter(paths: string[]): (x: any) => any {
  const segments = paths;
  let seg = 'store.' + segments[0],
    i = 0;
  const = segments.length;
  let expr = seg;
  while (++i < l) {
    expr = expr + ' && ' + (seg = seg + '.' + segments[i]);
  }
  const fn = new Function('store', 'return ' + expr + '');
  return <(x: any) => any>fn;
}
```
then 
```ts
if (typeof selectorOrFeature === 'string' && selectorOrFeature /* should throw error with empty string */) {
      let propsArray = [selectorOrFeature, ...paths];
      if (!paths.length) {
        propsArray = selectorOrFeature.split('.');
      }
      fn = fastPropGetter(propsArray);
    } else if (typeof selectorOrFeature === 'function') {
      fn = selectorOrFeature;
    } else {
       throw new Error('...')
    }
```

Reply:
we should check case when user put a dot in selectorOrFeature and paths.length > 0 too